xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -1068,10 +1069,9 @@ CDVDVideoCodec::VCReturn CDVDVideoCodecAndroidMediaCodec::GetPicture(VideoPictur
     else
     {
       m_videobuffer.videoBuffer = nullptr;
-      if (retgp == -1 || ((m_codecControlFlags & DVD_CODEC_CTRL_DRAIN)!=0 && ++m_noPictureLoop == 10))  // EOS
+      if (retgp == -1)  // EOS
```

Comment:
```
      else if (m_state == MEDIACODEC_STATE_WAIT_ENDOFSTREAM && retgp == -2)  // Error
      {
        m_state = MEDIACODEC_STATE_ERROR;
        return VC_EOF;
      }
```

if you don't consider error here, it can hang in an endless loop in case of error